wagtail/models/__init__.py

```diff
@@ -178,7 +178,15 @@ def get_streamfield_names(model_class):
 
 class BasePageManager(models.Manager):
     def get_queryset(self):
-        return self._queryset_class(self.model).order_by("path")
+        qs = self._queryset_class(self.model).order_by("path")
+        if self.model._meta.proxy:
+            # Because we store a `content_type` for every page, and proxy models have a
+            # unique `content_type`, we can automatically filter querysets requested from
+            # proxy models, so that they only include page saved from that model. Without
+            # this, all pages that share the concrete parent type would be included;
+            # which is generally less useful.
+            return qs.exact_type(self.model)
```

Comment:
What about the case where the proxy model is itself subclassed (either by another proxy model or MTI)? In that case I think we'd want the queryset to include instances of the subclass too, which `exact_type` won't do. However, I think simply changing this to `.type` will do the job (since it [explicitly looks for subclasses and then filters by that set of content_types](https://github.com/wagtail/wagtail/blob/5a249d4e9c65ec742c8e920d44f2fda2038395e3/wagtail/query.py#L200-L205)).

Reply:
Good spot. Thanks @gasman! Fixed in 0e0ad73